contracts/access/TimelockController.sol

```diff
@@ -0,0 +1,294 @@
+        bytes32 id = hashOperation(target, value, data, predecessor, salt);
+        _execute(id, predecessor);
+        _call(id, 0, target, value, data);
+    }
+
+    /**
+     * @dev Execute an (ready) operation containing a batch of transactions.
+     *
+     * Emits one {CallExecuted} event per transaction in the batch.
+     *
+     * Requirements:
+     *
+     * - the caller must have the 'executor' role.
+     */
+    function executeBatch(address[] calldata targets, uint256[] calldata values, bytes[] calldata datas, bytes32 predecessor, bytes32 salt) public payable onlyRole(EXECUTOR_ROLE) {
+        require(targets.length == values.length, "TimelockController: length missmatch");
+        require(targets.length == datas.length, "TimelockController: length missmatch");
+
+        bytes32 id = hashOperationBatch(targets, values, datas, predecessor, salt);
+        _execute(id, predecessor);
+        for (uint256 i = 0; i < targets.length; ++i) {
+            _call(id, i, targets[i], values[i], datas[i]);
+        }
+    }
+
+    /**
+     * @dev Execute an operation. Operation must be ready.
+     */
+    function _execute(bytes32 id, bytes32 predecessor) internal {
+        require(isOperationReady(id), "TimelockController: operation is not ready");
+        require(predecessor == bytes32(0) || isOperationDone(predecessor), "TimelockController: missing dependency");
+        _timestamps[id] = _DONE_TIMESTAMP;
+    }
+
+    /**
+     * @dev Execute a transaction.
+     *
+     * Emits a {CallExecuted} event.
+     */
+    function _call(bytes32 id, uint256 index, address target, uint256 value, bytes calldata data) internal returns (bool) {
+        // solhint-disable-next-line avoid-low-level-calls
+        (bool success,) = target.call{value: value}(data);
+        require(success, "TimelockController: underlying transaction reverted");
+
+        emit CallExecuted(id, index, target, value, data);
+    }
+
+    /**
+     * @dev Changes the timelock duration for future operations.
+     *
+     * Emits a {MinDelayChange} event.
+     */
+    function updateDelay(uint256 newDelay) external {
+        require(msg.sender == address(this), "TimelockController: restricted maintenance access");
+        emit MinDelayChange(newDelay, _minDelay);
+        _minDelay = newDelay;
+    }
+
+    /**
+     * @dev Revocate the sender's administrative power, and give this role to
+     * the timelock itself. All further maintenance will have to be performed
+     * by the timelock itself using the commit/reveal workflow.
+     *
+     * Emits one {RoleGranted} and one {RoleRevoked} event.
+     *
+     * Requirements:
+     *
+     * - the caller must be the only address with `ADMIN_ROLE`.
+     * - there must be at least one account with 'PROPOSER_ROLE' and one with
+     *   'EXECUTOR_ROLE'.
+     */
+    function makeLive() external /* onlyRole(ADMIN_ROLE) */ {
```

Comment:
I've been thinking about this function and I think we should remove it. Although we think the pattern makes sense, the name `makeLive` implies that we see it as a requirement for a timelock to be "right". But I think that will depend on the team. If a team wants to implement this pattern, they are free to do so, but we shouldn't force them.

Maybe the timelock should have `ADMIN_ROLE` by default assigned in the constructor, and the pattern would be reduced to renouncing the role from the deployer contract.

Reply:
`makeLive` has been removed